lupa/lock.pxi

```diff
@@ -12,13 +23,14 @@ cdef class FastRLock:
     wonderful GIL.
     """
     cdef pythread.PyThread_type_lock _real_lock
-    cdef long _owner            # ID of thread owning the lock
+    cdef bint _has_owner        # whether the lock has an owner
+    cdef pythread_t _owner      # ID of thread owning the lock
     cdef int _count             # re-entry count
     cdef int _pending_requests  # number of pending requests for real lock
     cdef bint _is_locked        # whether the real lock is acquired
 
     def __cinit__(self):
-        self._owner = -1
+        self._has_owner = False
```

Comment:
Use `<pythread_t> -1` as sentinel, maybe?

Reply:
Not really safe, because it's still a valid thread id.